File: src/main/java/cz/linkedlist/DownInfoService.java
package cz.linkedlist;

import org.springframework.util.concurrent.ListenableFuture;

/**
 * @author Martin Macko <https://github.com/LinkedList>
 */
public interface DownInfoService {

    /**
     * Will download TileInfo for specified tileSet
     * @param tileSet
     * @return new instance of TileSet with downloaded TileInfo
     */
    ListenableFuture<TileSet> downTileInfo(final TileSet tileSet);
}

File: src/main/kotlin/cz/linkedlist/DownInfoService.kt
package cz.linkedlist

import org.springframework.util.concurrent.ListenableFuture

/**
 * @author Martin Macko <https:></https:>//github.com/LinkedList>
 */
interface DownInfoService {

    /**
     * Will download TileInfo for specified tileSet
     * @param tileSet
     * @return new instance of TileSet with downloaded TileInfo
     */
    fun downTileInfo(tileSet: TileSet): ListenableFuture<TileSet>
}
